.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,48 @@
+name: Prepare Release
+
+on: workflow_dispatch
```

Comment:
Can we also add a prefix to the release?  We sometimes do `-beta` or `-rc` releases, such as:
`11.0.0-rc.1`

Reply:
Hmm there is currently support for a "tag" in the Publish Release step. It gets passed along in the npm publish step. It defaults to "latest", and if it's not "latest", then the release is marked as a prerelease on github.

Would it make sense to use that tag in the version as well? Both in the github release and in the git tag for the release. But not in the version passed to npm publish, since it already has "tag" there.

Then I would also make it so that it doesn't merge the PR if the tag is not "latest', because you probably want to use the same branch for a real release later.

Reply:
Okay :)
I can add a "prerelease" variable next to "tag", default empty. And then handle the version logic based on that.

I would guess the 11.0.0 prereleases are made from a different branch than master? Since presumably they contain breaking changes.  (We also still need to handle the case of releasing from a base branch other than master)

So I still think it's best if the prerelease string is set in the "publish release" step, so that you can do one "prepare release" that creates a release branch, then optionally multiple prerelease "publish release" steps from there (which don't merge the branch into master), then a final "publish release" when the release is ready.

But if you want it the other way around, where each prerelease gets its own release branch named with the prerelease tag and a new one is made for every prerelease, let me know.

Btw, if you want to merge this PR soon then I can optionally work on the prerelease feature as a new PR. Since it's kind of orthogonal to the current PR.

Reply:
Yes, let's do this at a later time.  One can always change the PR prepared by the workflow anyway.